### PR TITLE
remove most references to currentWindow and lazy load the rest

### PR DESCRIPTION
**Pull request by bridiver**
check for automated tests to pass before merging

### CODE REVIEW

**Comment by bridiver** [comment]
updated with unit test fix

File: app/renderer/components/navigation/navigator.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ class Navigator extends ImmutableComponent {
     if (!e.target.className.includes('navigatorWrapper')) {
       return
     }
-    return !isMaximized() ? currentWindow.maximize() : currentWindow.unmaximize()
+    return !isMaximized() ? windowActions.shouldMaximize(getCurrentWindowId()) : windowActions.shouldMinimize(getCurrentWindowId())
```
Comment (bbondy) [inline app/renderer/components/navigation/navigator.js]: I guess should* is kind of like *Requested for tab creation, maybe we should use one or the other.  No need to hold this up though just thinking out loud.

**Comment by bbondy** [comment]
++

File: app/renderer/components/navigation/navigator.js (hunk 1 above)
Comment (bridiver) [inline app/renderer/components/navigation/navigator.js]: these are really just temporary to replace the currentWindow.method calls. This should really be `onNavigatorDoubleClick` or something like that

Comment (bridiver) [inline app/renderer/components/navigation/navigator.js]: see https://github.com/brave/browser-laptop/pull/8220/files#diff-2db1344f65c9be10474c5562bcef4c60R1134

**Pull request merged**